source/linux/epoll_event_loop.c

```diff
@@ -576,6 +584,8 @@ static void aws_event_loop_thread(void *args) {
         return;
     }
 
+    aws_thread_current_at_exit(s_aws_cleanup_aws_lc_thread_local_state, NULL);
```

Comment:
oye.... what's the right place to call this from?
You picked epoll because we use that on Linux
But BSD uses kqueue

`s2n_cleanup()` is similar in that you need to call it on any thread that interacts with S2N, to avoid leaks...
[Here](https://github.com/awslabs/aws-c-io/blob/c3f95ebf3d6d1608e1656bb08eca05e303cc4fd3/source/s2n/s2n_tls_channel_handler.c#L1146) is where the s2n_channel_handler registers that to run on any event-loop thread running S2N. We could add the aws-lc call here too? because anywhere we use AWS-LC we're also using S2N?

Or we could be super blunt and just have **EVERY** aws_thread call this ...

Still doesn't cover misc user threads from other languages that might call `aws_tls_ctx_new()` ...

Reply:
You could sign in an event loop thread that isn't running a connection so I think all event loop threads should unconditionally clean up, not just ones that s2n has touched.